File: benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/FillReceiveQueuePhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

class FillReceiveQueuePhase extends BenchmarkPhase {

    FillReceiveQueuePhase(Benchmark benchmark) {
        super(benchmark);
    }

    @Override
    Status execute() {
        return getBenchmark().fillReceiveQueue();
    }
}

File: cpp/src/benchmark/phase/FillReceiveQueuePhase.cpp
#include "FillReceiveQueuePhase.h"
#include <benchmark/Benchmark.h>

#include <utility>
#include <benchmark/operation/Operation.h>

namespace Observatory {

FillReceiveQueuePhase::FillReceiveQueuePhase(Benchmark &benchmark, const Operation &operation) :
        BenchmarkPhase(benchmark),
        fillReceiveQueue(operation.needsFilledReceiveQueue()) {}

const char* FillReceiveQueuePhase::getName() {
    return "FillReceiveQueuePhase";
}

Status FillReceiveQueuePhase::execute() {
    if(fillReceiveQueue) {
        return getBenchmark().fillReceiveQueue();
    }

    return Status::OK;
}

}